### PR TITLE
Deprecate local handling of memref.extract_strided_metadata in favor of upstream patterns.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**
The initial plumbing of micro kernels adapted the patterns in `ResolveBufferDescriptor` to use
`memref.extract_strided_metadata`. With #12874 now these patterns can
be deprecated in favor of those upstream. The pass itself can be
deprecated after the legacy micro kernel paths are all ported over to
the tensor based `ukernel.*` ops.
    
This also makes the `iree_codegen.get_base_pointer` operation
unnecessary. The op and corresponding transformations are removed as
well.
    
Depends on https://reviews.llvm.org/D147393

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Common/FlattenMemRefSubspanPass.cpp [hunk 1]
```diff
@@ -292,6 +292,32 @@ struct FlattenBindingSubspan final
   }
 };
 
+/// Flatten `memref` operands and results of `memref.reinterpret_cast` op.
```
Comment (antiagainst) [inline compiler/src/iree/compiler/Codegen/Common/FlattenMemRefSubspanPass.cpp:295]: Can we add a test for this case?

File: compiler/src/iree/compiler/Codegen/Common/IREEExpandStridedMetadata.cpp [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+  }
+  AffineMap mulMap = getMulMap(rewriter.getContext());
+  for (int i = sizes.size() - 2; i >= 0; --i) {
+    strides[i] = makeComposedFoldedAffineApply(rewriter, loc, mulMap,
+                                               {strides[i + 1], sizes[i + 1]});
+  }
+  return strides;
+}
+
+static FailureOr<DescriptorInfo> resolveBufferDescriptorForInterfaceBinding(
```
Comment (antiagainst) [inline compiler/src/iree/compiler/Codegen/Common/IREEExpandStridedMetadata.cpp:63]: This function won't fail? Don't need to have the wrapping `FailureOr` to me.

**Review by antiagainst — APPROVED** [review]
LGTM!

File: compiler/src/iree/compiler/Codegen/Common/IREEExpandStridedMetadata.cpp (hunk 2 above)
Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Codegen/Common/IREEExpandStridedMetadata.cpp:63]: I'd prefer to keep it that way... It doesnt fail today, but just keeping it this way to make sure any failures do get propagated.

**Pull request merged**